zipline/pipeline/loaders/earnings_estimates.py

```diff
@@ -651,3 +825,700 @@ def get_zeroth_quarter_idx(self, stacked_last_per_qtr):
+                    # until that KD.
+                    end_idx = self.determine_end_idx_for_adjustment(
+                        timestamp,
+                        requested_qtr_data.index,
+                        upper_bound,
+                        requested_quarters_per_range[i],
+                        sid_estimates
+                    )
+                    # In the default case, apply adjustment on the first day of
+                    #  the quarter.
+                    start_idx = qtr_range[0]
+                    # If the adjustment happens during this quarter, apply the
+                    # adjustment on the day it happens.
+                    if date_index > start_idx:
+                        start_idx = date_index
+                    # We only want to apply the adjustment if we have any stale
+                    # data to apply it to.
+                    if qtr_range[0] <= end_idx:
+                        for column_name in requested_split_adjusted_columns:
+                            if column_name not in col_to_split_adjustments:
+                                col_to_split_adjustments[column_name] = {}
+                            adj = Float64Multiply(
+                                # Always apply from first day of qtr
+                                qtr_range[0],
+                                end_idx,
+                                sid_idx,
+                                sid_idx,
+                                adjustment
+                            )
+                            add_new_adjustments(
+                                col_to_split_adjustments,
+                                [adj],
+                                column_name,
+                                start_idx
+                            )
+
+        return col_to_split_adjustments
+
+    def retrieve_split_adjustment_data_for_sid(self,
+                                               dates,
+                                               sid,
+                                               split_adjusted_asof_idx):
+        """
+        dates : pd.DatetimeIndex
+            The calendar dates.
+        sid : int
+            The sid for which we want to retrieve adjustments.
+        split_adjusted_asof_idx : int
+            The index in `dates` as-of which the data is split adjusted.
+
+        Returns
+        -------
+        pre_adjustments : tuple(list(float), list(int), pd.DatetimeIndex)
+            The adjustment values and indexes in `dates` for
+            adjustments that happened before the split-asof-date.
+        pre_adjustments : tuple(list(float), list(int), pd.DatetimeIndex)
+            The adjustment values, indexes in `dates`, and timestamps for
+            adjustments that happened after the split-asof-date.
+        """
+        adjustments = self._split_adjustments.get_adjustments_for_sid(
+            'splits', sid
+        )
+        sorted(adjustments, key=lambda adj: adj[0])
```

Comment:
I believe `get_adjustments_for_sid` always returns in order (but may not be explicitly defined in the adjustment reader interface). However, since any given amount of splits is small, ensuring that they are in order here should be ok.